src/AbstractEventStateMachine.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace RebelCode\State;
+
+use Dhii\Events\TransitionEventInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception;
+use Psr\EventManager\EventManagerInterface;
+
+abstract class AbstractEventStateMachine
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    protected function _transition($transition)
+    {
+        $event = $this->_getTransitionEvent($transition);
+
+        try {
+            $this->_getEventManager()->trigger($event);
+        } catch (Exception $ex) {
+            $exception = $this->_createCouldNotTransitionException(
+                $this->__('The triggered event threw an exception'),
+                null,
+                $ex,
+                $transition
+            );
+        }
+
+        if (!$event->isTransitionAborted()) {
+            $this->_setState($this->_getNewState($event));
+        }
+
+        if (isset($exception) && $exception instanceof Exception) {
+            throw $exception;
```

Comment:
Nothing is apparently thrown if event handlers did not throw. However, the interface, AFAIK, [requires](https://github.com/Dhii/machine-state-machine-interface/blob/develop/src/StateMachineInterface.php#L31) one of two things happen: the transition happens, or [`CouldNotTransitionExceptionInterface`] is thrown. Therefore, the exception must be thrown when the transition is aborted, regardless of whether the event mechanism threw.

[`CouldNotTransitionExceptionInterface`]: https://github.com/Dhii/machine-state-machine-interface/blob/develop/src/Exception/CouldNotTransitionExceptionInterface.php